feature-libs/cart/main/core/http-interceptors/handlers/bad-cart-request.handler.ts

```diff
@@ -0,0 +1,49 @@
+import { HttpErrorResponse, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import {
+  ErrorModel,
+  GlobalMessageType,
+  HttpErrorHandler,
+  HttpResponseStatus,
+  Priority,
+} from '@spartacus/core';
+import { isCartNotFoundError } from '../../utils/utils';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BadCartRequestHandler extends HttpErrorHandler {
+  responseStatus = HttpResponseStatus.BAD_REQUEST;
+
+  getPriority(): Priority {
+    return Priority.NORMAL;
+  }
+
+  hasMatch(errorResponse: HttpErrorResponse): boolean {
+    return this.getErrors(errorResponse).some(isCartNotFoundError);
```

Comment:
better this: `super.hasMatch(errorResponse) && this.getErrors(errorResponse).some(isCartNotFoundError)`

Reply:
Fixed.  Good catch.